### ISSUE TITLE
Scores on the demo

### ISSUE THREAD

**Issue by koraykoca** [tianweiy/CenterPoint-KITTI#35]
Hi, I have some questions if you have time:

1- As written in the paper, CenterPoint gives scores and 3D boxes after the second stage. As written here in the issues, CenterPoint-KITTI doesn't have the second stage. I wonder, how can I see scores and 3D boxes in the demo? How/Where are these scores calculated in the code? Is it calculated as IoU?

2- General question: Points have (x, y, z, intensity) values, then we voxelized them. If I understand it right, here voxelization means actually dividing the whole point cloud into small sub-volumes with size e.g. (0.05, 0.05, 0.1), and each volume (voxel) has e.g. 5 points. Then we take the average of points in each volume and represent that volume as one averaged point. After that, we again have (x,y,z, intensity) values but averaged. Is my understanding right? And what is the actual purpose here? Why do we need to do this?

Thank you in advance.

**Comment by tianweiy** [comment]
1. i never used the demo file. but the scores will be computed here https://github.com/tianweiy/CenterPoint-KITTI/blob/222232e5554cbe6e3d19d90faa8e1be4fd11fd3d/pcdet/models/detectors/detector3d_template.py#L272 it is not iou (simply the heatmap prediction score)

2. you are right. In this way, we turn voxels with different number of points to a single feature vector which we can pass into networks for predictions

**Comment by koraykoca** [comment]
Thank you for your fast answer. 

a follow-up question: Does "voxels with the different number of points" mean voxels can include between 0 and 5 number of points (max. 5 points, we decided it), right? So if for example, when 10 points fall into a voxel, then we remove the 5 of them?

**Comment by tianweiy** [comment]
yes, we will zero pad but during average of features we will only compute average for valid points. 

So if for example, when 10 points fall into a voxel, then we remove the 5 of them?
yes

**Comment by koraykoca** [comment]
thank you @tianweiy for your time. I have one more question;

> it is not iou (simply the heatmap prediction score)
Does simply the heatmap prediction score mean that we have probability values from 0 to 1 in the heat map and 1 is ground truth, so if our peak value for an object in the heat map is 0.9, then we say we predict this object with the 90% score/confidence?

**Comment by TumVink** [comment]
Heyy Koca,

As the paper said,  "confidence is proportional to the heatmap value at the peak". So i guess it may not be exactly 90%.

Jingsong
From course ADL4CV ;)
